snap/info_snap_yaml.go

```diff
@@ -303,6 +315,67 @@ func infoSkeletonFromSnapYaml(y snapYaml) *Info {
 	return snap
 }
 
+func setComponentsFromSnapYaml(y snapYaml, snap *Info, strk *scopedTracker) {
+	if len(y.Components) > 0 {
+		snap.Components = make(map[string]*Component, len(y.Components))
+	}
+
+	for name, data := range y.Components {
+		var hooks map[string]*HookInfo
+		if len(data.Hooks) > 0 {
+			hooks = make(map[string]*HookInfo, len(data.Hooks))
+		}
+
+		for hookName, hookData := range data.Hooks {
+			if !IsComponentHookSupported(hookName) {
+				continue
```

Comment:
Souldn't we return an error here instead? We don't for normal hooks but maybe we should be more strict with this as we don't have to think about backwards compatibility.

Reply:
Makes sense, I didn't know that was because of backwards compatibility.